src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs

```diff
@@ -41,7 +41,7 @@ private sealed class AnalyzerContext(Compilation compilation)
         private INamedTypeSymbol? ResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Http.IResult");
         private INamedTypeSymbol? ActionResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Mvc.IActionResult");
 
-        public bool IsValid => this.ActionResultSymbol is not null && this.ResultSymbol is not null;
+        public bool IsValid => this.ActionResultSymbol is not null || this.ResultSymbol is not null;
```

Comment:
Add test to prevent regression

Reply:
This caused an error because it was making sure that the file has both `using Microsoft.AspNetCore.Http;` and
`using Microsoft.AspNetCore.Mvc;`. Given that we extend the Analyzer tests with BaseAnalyzer which provided both sources in the same list, we weren't able to catch it. 

I've added an extra `this.TestState.Sources.Add(AspNetCoreHttpUsing);` on the two tests requiring IResult and moved it out of the default using list.